ompi/mca/mtl/psm2/mtl_psm2.c

```diff
@@ -101,7 +103,17 @@ int ompi_mtl_psm2_module_init(int local_rank, int num_local_procs) {
     char env_string[256];
     int rc;
 
-    generated_key = getenv("OMPI_MCA_orte_precondition_transports");
+    opal_process_name_t pname;
+
+    generated_key = NULL;
+    pname.jobid = opal_process_info.my_name.jobid;
+    pname.vpid = OPAL_VPID_WILDCARD;
+    OPAL_MODEX_RECV_VALUE_OPTIONAL(rc, PMIX_CREDENTIAL, &pname,
+                                    (char**)&generated_key, PMIX_STRING);
+
+    if (PMIX_SUCCESS != rc || NULL == generated_key) {
+        generated_key = getenv("OMPI_MCA_orte_precondition_transports");
```

Comment:
You need to "strdup" the `getenv` value here (checking for NULL first) so you can `free` it later

Reply:
Thanks